divi/src/main.cpp

```diff
@@ -673,7 +673,7 @@ bool IsStandardTx(const CTransaction& tx, string& reason)
     // Timestamps on the other hand don't get any special treatment, because we
     // can't know what timestamp the next block will have, and there aren't
     // timestamp applications where it matters.
-    if (!IsFinalTx(tx, chainActive.Height() + 1)) {
+    if (!IsFinalTx(tx, chainActive, chainActive.Height() + 1)) {
```

Comment:
I think you missed a few '_' for some of the instances that should be the wallet's new private member references to the active chain and mapBlockIndex, but otherwise most of this looks good! Awesome!

Reply:
Hm, interesting.  I thought that I actually removed the declarations of the globals (so it wouldn't compile in that case), but will investigate this a bit more in depth then.

Reply:
Ok, I now confirmed that the code in `wallet.cpp` definitely doesn't compile if I try to reference the global `chainActive` or `mapBlockIndex` (which also means that all references to them are correctly to the local reference).

The line in question here (as well as some others in the PR) is not in the wallet, and thus will remain referencing the global.  It is needed because I changed also the signature of `IsFinalTx` to explicitly accept the active chain as argument, since I need that from the wallet code.  Hence I had to update non-wallet callers as well, of which this line is one.  I think all should be good.